parquet-variant/src/builder.rs

```diff
@@ -626,23 +689,33 @@ impl ParentState<'_> {
 /// let (metadata, value) = builder.finish();
 /// let variant = Variant::try_new(&metadata, &value).unwrap();
 /// ```
-///
-#[derive(Default)]
+#[derive(Default, Debug)]
 pub struct VariantBuilder {
     buffer: ValueBuffer,
     metadata_builder: MetadataBuilder,
     validate_unique_fields: bool,
 }
 
 impl VariantBuilder {
+    /// Create a new VariantBuilder with new underlying buffer
     pub fn new() -> Self {
         Self {
-            buffer: ValueBuffer::default(),
+            buffer: ValueBuffer::new(),
             metadata_builder: MetadataBuilder::default(),
             validate_unique_fields: false,
         }
     }
 
+    /// Create a new VariantBuilder that will write the metadata and values to
```

Comment:
```suggestion
    /// Create a new VariantBuilder that will append the metadata and values to
```
(good to be clear that this won't erase existing content)